server/lib/voice_parser.py

```
@@ -0,0 +1,156 @@
+            response_mime_type="application/json",
+            response_schema=VoiceParseResult,
+        ),
+    )
+
+    parsed: VoiceParseResult | None = getattr(response, "parsed", None)
+    if parsed is None:
+        try:
+            parsed = VoiceParseResult.model_validate_json(response.text or "")
+        except ValidationError as e:
+            logger.warning("Voice parser returned unparseable response: %s", e)
+            raise RuntimeError("Gemini response did not match VoiceParseResult schema.") from e
```

Ruff (0.15.12)

[warning] Avoid specifying long messages outside the exception class

(TRY003)